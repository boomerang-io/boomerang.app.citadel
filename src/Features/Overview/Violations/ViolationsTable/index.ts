// @ts-expect-error ts-migrate(6142) FIXME: Module './ViolationsTable' was resolved to 'C:/Use... Remove this comment to see the full error message
export { default } from "./ViolationsTable"