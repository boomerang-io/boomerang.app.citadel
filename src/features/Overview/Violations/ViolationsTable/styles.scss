@import "~@boomerang/boomerang-styles/lib/colors";
@import "~Styles/variables";

.c-violations-table {
  margin-top: 1rem;
}

.b-violations-table-head__row {
  background-color: $cool-white-1;
}

.b-violations-table-head__row > .b-violations-table-head__header {
  color: $cerulean-40;
  font-size: inherit;
  font-family: inherit;
  font-weight: 600;
  height: 3rem;

  .bx--table-header-label {
    color: $cerulean-50;
    font-size: 0.9rem;
    font-weight: bolder;
    padding-right: 0.5rem;
  }

  &.--policyName {
    width: 20%;
  }
  &.--stage {
    width: 20%;
  }
  &.--componentName {
    width: 20%;
  }
  &.--componentName {
    width: 20%;
  }
  &.--reason {
    width: 20%;
  }
}

.b-violations-table-body {
  background-color: #f2f2f2;
  color: $gray-90;
  font-weight: 400;
}

.b-violations-table-body__cell {
  align-items: center;
  display: flex;
  height: 3.5rem;
  padding-left: 0.75rem;
  width: 100%;
}
.c-violations-table-body__icons {
  display: flex;
  justify-content: flex-end;
  width: 100%;
}
.b-violations-table-body__icon {
  cursor: pointer;
  display: flex;
  height: 1.25rem;
  margin-right: 1.5rem;
  width: 1.25rem;
}

.b-violations-table-body__textarea {
  background: inherit;
  color: $black;
  max-width: 100%;
  resize: none;
  text-align: left;
  white-space: normal;
  width: 90%;
  margin-left: 0.25rem;
}
