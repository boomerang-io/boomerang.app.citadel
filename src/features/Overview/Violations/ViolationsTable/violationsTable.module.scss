@import "~Styles/variables";

.tableContainer {
  margin-top: 1rem;
}

.tableHeadRow {
  //background-color: $cool-white-1;
}

.tableHeadRow > .tableHeader {
  color: $interactive-01;
  font-size: inherit;
  font-family: inherit;
  font-weight: 600;
  height: 3rem;

  &.ciPolicyName {
    width: 10%;
  }

  &.ciStageName {
    width: 7.5%;
  }

  &.ciComponentName {
    width: 15%;
  }

  &.ciComponentVersionName {
    width: 10%;
  }

  &.violations {
    width: 5%;
  }

  &.ciPolicyDefinitionTypes {
    width: 25%;
    overflow-y: auto;
  }

  &.ciPolicyActivityCreatedDate {
    width: 12%;
  }
}
.tableSubHeader {
  color: $interactive-01;
  font-size: inherit;
  font-family: inherit;
  font-weight: 600;
  height: 3rem;
  margin-left: 3.5rem;

  &.metric {
    width: 30%;
  }
  &.message {
    width: 70%;
  }
}

.tableBody {
  background-color: #f2f2f2;
  //color: $gray-90;
  font-weight: 400;
  :global .bx--expandable-row {
    td {
      padding: 1rem 0rem !important;
    }
  }
}

.tableCell {
  align-items: center;
  display: flex;
  height: 3.5rem;
  line-height: 1.2rem;
  padding-left: 0.75rem;
  width: 100%;
  overflow-y: auto;
  &.metric {
    width: 30%;
  }
  &.message {
    width: 70%;
  }
}
.tableTextarea {
  background: inherit;
  color: $text-01;
  max-width: 100%;
  max-height: 100%;
  resize: none;
  text-align: left;
  white-space: normal;
  width: 90%;
  margin-left: 0.25rem;
  word-break: break-all;
}
.tableSubHeaders {
  display: flex;
  width: 100%;
}
.detailMetrics {
  //color: $gray-90;
  margin-left: 3rem;
  width: 100%;
}
.detailMessage {
  // color: $gray-50;
  margin-left: 4.25rem;
  width: 100%;
}
.subRow {
  display: flex;
  width: 100%;
  &:hover {
    //background-color: $gray-1;
  }
}
.noDetail {
  margin-left: 3.25rem;
}
