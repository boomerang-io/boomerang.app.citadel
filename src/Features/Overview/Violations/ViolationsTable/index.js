import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import moment from "moment";
import {
  Button,
  CodeSnippet,
  ComposedModal,
  DataTable,
  ModalFooter,
  ModalForm,
  ModalBody,
  Tag,
} from "@boomerang-io/carbon-addons-boomerang-react";
import copy from "copy-to-clipboard";
import { isAccessibleEvent } from "@boomerang-io/utils";
import styles from "./violationsTable.module.scss";

export class ViolationsTable extends Component {
  static propTypes = {
    violations: PropTypes.array,
  };

  annotationsRef = React.createRef();

  state = { isModalOpen: false, selectedPolicyIndex: null };
  headers = [
    {
      header: "Policy",
      key: "policyName",
    },
    {
      header: "Violations",
      key: "nbrViolations",
    },
    {
      header: "Failed Definition Types",
      key: "policyDefinitionTypes",
    },
    {
      header: "Reference ID",
      key: "referenceId",
    },
    {
      header: "Activity Date",
      key: "policyActivityCreatedDate",
    },
  ];

  subHeaders = [
    {
      header: "Metrics",
      key: "metric",
    },
    {
      header: "Message",
      key: "message",
    },
  ];

  renderCell = (rowIndex, cellIndex, value) => {
    const column = this.headers[cellIndex];
    switch (column.header) {
      case "Activity Date":
        return <time className={styles.tableTextarea}>{value ? moment(value).format("MMMM DD, YYYY") : "---"}</time>;
      case "Failed Definition Types":
        return <p className={styles.tableTextarea}>{value && value.length ? value.join(", ") : "---"}</p>;
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  renderSubRow = (row) => {
    const { violations } = this.props;
    const currentViolation = violations.find((violation) => violation.id === row.id);
    if (currentViolation && currentViolation.violations.length > 0)
      return currentViolation.violations.map((violation) => (
        <div className={styles.subRow}>
          {this.subHeaders.map((cell) => (
            <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
              {this.renderDetail(cell.key, violation[cell.key])}
            </div>
          ))}
        </div>
      ));
    else {
      return (
        <div className={styles.subRow}>
          {this.subHeaders.map((cell) => (
            <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
              {this.renderDetail(cell.key, "---")}
            </div>
          ))}
        </div>
      );
    }
  };

  renderDetail = (key, value) => {
    switch (key) {
      case "metric":
        return <p className={styles.detailMetrics}>{value}</p>;
      case "message":
        return <p className={styles.detailMessage}>{value}</p>;
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  handleRowClick = (row) => {
    this.setState({
      isModalOpen: true,
      selectedPolicyIndex: row,
    });
  };

  render() {
    const { violations } = this.props;
    const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;
    const selectedViolation = violations[this.state.selectedPolicyIndex];
    return (
      <>
        <ComposedModal
          isOpen={this.state.isModalOpen}
          composedModalProps={{ shouldCloseOnOverlayClick: true }}
          onCloseModal={() => this.setState({ isModalOpen: false })}
          modalHeaderProps={{ title: selectedViolation?.policyName }}
        >
          {() => (
            <ModalForm>
              <ModalBody>
                <h2 className={styles.modalSectionTitle}>{`Violations (${
                  selectedViolation?.violations?.length ?? 0
                })`}</h2>
                <ul className={styles.modalLabels}>
                  {(selectedViolation?.violations ?? [])?.map(({ metric, message }, index) => (
                    <li key={index}>
                      <span>{metric} : </span>
                      <span style={{ fontStyle: "italic" }}>{message}</span>
                    </li>
                  ))}
                </ul>
                <h2 className={styles.modalSectionTitle}>Failed Definition Types</h2>
                <p>
                  {selectedViolation?.policyDefinitionTypes && selectedViolation?.policyDefinitionTypes.length
                    ? selectedViolation?.policyDefinitionTypes.join(", ")
                    : "---"}
                </p>
                <h2 className={styles.modalSectionTitle}>Labels</h2>
                <ul className={styles.modalLabels}>
                  {Object.entries(selectedViolation?.labels ?? {})?.map((entry) => (
                    <li key={entry[0]}>
                      <Tag type="purple">
                        {entry[0]}
                        <span>:</span>
                        {entry[1]}
                      </Tag>
                    </li>
                  ))}
                </ul>
                <h2 className={styles.modalSectionTitle}>Reference Link</h2>
                {selectedViolation?.referenceLink ? (
                  <a href={selectedViolation.referenceLink} alt="Reference link">
                    {selectedViolation.referenceLink}
                  </a>
                ) : (
                  <p>No reference link provided</p>
                )}

                <h2 className={styles.modalSectionTitle}>Reference ID</h2>
                <p>{selectedViolation?.referenceId ?? "---"}</p>
                <h2 className={styles.modalSectionTitle}>Created Date</h2>
                <time>
                  {selectedViolation?.createdDate
                    ? moment(selectedViolation.createdDate).format("MMMM DD, YYYY")
                    : "---"}
                </time>
                <h2 className={styles.modalSectionTitle}>Annotations</h2>
                {selectedViolation?.annotations ? (
                  <CodeSnippet
                    copyButtonDescription="Copy annotations to clipboard"
                    onClick={() => copy(JSON.stringify([selectedViolation?.annotations]))}
                    type="multi"
                  >
                    {JSON.stringify([selectedViolation?.annotations], null, 1)}
                  </CodeSnippet>
                ) : (
                  "---"
                )}
              </ModalBody>
              <ModalFooter>
                <Button onClick={() => this.setState({ isModalOpen: false })} data-testid="close-violation-detail">
                  Close
                </Button>
              </ModalFooter>
            </ModalForm>
          )}
        </ComposedModal>
        <DataTable
          rows={violations}
          headers={this.headers}
          isSortable={true}
          render={({ rows, headers, getHeaderProps, getRowProps }) => (
            <TableContainer>
              <Table className={styles.tableContainer} sortable={"true"} useZebraStyles={false}>
                <TableHead>
                  <TableRow className={styles.tableHeadRow}>
                    {headers.map((header) => (
                      <TableHeader
                        {...getHeaderProps({ header, className: `${styles.tableHeader} ${styles[header.key]}` })}
                      >
                        {header.header}
                      </TableHeader>
                    ))}
                  </TableRow>
                </TableHead>
                <TableBody className={styles.tableBody}>
                  {rows.map((row, rowIndex) => (
                    <Fragment key={row.id}>
                      <TableRow
                        className={styles.tableBodyRow}
                        key={row.id}
                        {...getRowProps({ row })}
                        onClick={() => this.handleRowClick(rowIndex)}
                        onKeyDown={(e) => isAccessibleEvent(e) && this.handleRowClick(rowIndex)}
                        tabIndex={0}
                        data-testid="violations-table-row"
                      >
                        {row.cells.map((cell, cellIndex) => (
                          <TableCell key={cell.id} style={{ padding: "0" }}>
                            <div className={styles.tableCell}>{this.renderCell(rowIndex, cellIndex, cell.value)}</div>
                          </TableCell>
                        ))}
                      </TableRow>
                    </Fragment>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
          )}
        />
      </>
    );
  }
}

export default ViolationsTable;
