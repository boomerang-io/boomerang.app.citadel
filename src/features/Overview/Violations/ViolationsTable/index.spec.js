import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import ViolationsTable from "./index";

const violations = [
  {
    "id":"1",
    "ciPolicyId": "5cd49adff6ea74a9bb6adef4",
    "ciPolicyName": "Tyson's Policy",
    "ciStageId": "5cb123bdf6ea74a9bb5425b6",
    "ciStageName": "dev",
    "ciComponentId": "5cdb4c15f6ea74a9bb9cfdba",
    "ciComponentName": "boomerang.service.ci",
    "ciComponentVersionId": "5cdb4ca3f6ea74a9bb9cf251",
    "ciComponentVersionName": "6.4.9-175",
    "ciPolicyActivityCreatedDate": 1558453998666,
    "violations": 3
  }
];

describe("ViolationsTable --- Snapshot", () => {
  it("Capturing Snapshot of ViolationsTable", () => {
    const renderedValue = renderer
      .create(
        <ViolationsTable
          violations={violations}
        />
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("ViolationsTable --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ViolationsTable
        violations={violations}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
