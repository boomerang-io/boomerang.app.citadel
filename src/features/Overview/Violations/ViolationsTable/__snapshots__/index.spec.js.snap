// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViolationsTable --- Snapshot Capturing Snapshot of ViolationsTable 1`] = `
<div
  className="bx--data-table-container"
>
  <table
    className="bx--data-table tableContainer bx--data-table--no-border"
    sortable="true"
  >
    <thead>
      <tr
        className="tableHeadRow"
      >
        <th
          className="bx--table-expand"
          scope="col"
        />
        <th
          aria-sort="none"
          className="tableHeader ciPolicyName"
          scope="col"
        >
          <button
            className="tableHeader ciPolicyName bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Policy
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader ciStageName"
          scope="col"
        >
          <button
            className="tableHeader ciStageName bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Stage
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader ciComponentName"
          scope="col"
        >
          <button
            className="tableHeader ciComponentName bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Component
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader ciComponentVersionName"
          scope="col"
        >
          <button
            className="tableHeader ciComponentVersionName bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Version
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader nbrViolations"
          scope="col"
        >
          <button
            className="tableHeader nbrViolations bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Violations
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader ciPolicyDefinitionTypes"
          scope="col"
        >
          <button
            className="tableHeader ciPolicyDefinitionTypes bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Failed Definition Types
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
        <th
          aria-sort="none"
          className="tableHeader ciPolicyActivityCreatedDate"
          scope="col"
        >
          <button
            className="tableHeader ciPolicyActivityCreatedDate bx--table-sort"
            onClick={[Function]}
          >
            <span
              className="bx--table-header-label"
            >
              Activity Date
            </span>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4L6 14l1.41 1.41L15 7.83V30h2V7.83l7.59 7.58L26 14 16 4z"
              />
            </svg>
            <svg
              aria-label="Sort rows by this header in ascending order"
              className="bx--table-sort__icon-unsorted"
              focusable="false"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26.59 20.59L23 24.17V4h-2v20.17l-3.59-3.58L16 22l6 6 6-6-1.41-1.41zM10 4l-6 6 1.41 1.41L9 7.83V28h2V7.83l3.59 3.58L16 10l-6-6z"
              />
            </svg>
          </button>
        </th>
      </tr>
    </thead>
    <tbody
      aria-live="polite"
      className="tableBody"
    >
      <tr
        className="bx--parent-row"
        data-parent-row={true}
        disabled={false}
      >
        <td
          className="bx--table-expand"
          headers="expand"
        >
          <button
            aria-label="Expand current row"
            className="bx--table-expand__button"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="bx--table-expand__svg"
              focusable="false"
              height={16}
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 8l-5 5-.7-.7L9.6 8 5.3 3.7 6 3z"
              />
            </svg>
          </button>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              Tyson's Policy
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              dev
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              boomerang.service.ci
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              6.4.9-175
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              ---
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              ---
            </p>
          </div>
        </td>
        <td
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <div
            className="tableCell"
          >
            <p
              className="tableTextarea"
            >
              Jan 01, 2019 - 12:00 am
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
